lib/shared/sse-connection/src/lib/SSEConnection.ts

```diff
@@ -20,15 +27,17 @@ export class SSEConnection {
         }
         this.connection = new EventSource(this.url, { withCredentials: true })
         this.connection.onmessage = (event) => {
-            this.onMessage(event.data)
+            this.callbacks.onMessage(event.data)
         }
         this.connection.onerror = () => {
             this.logger.warn(
                 'StreamingConnection warning. Connection failed to establish.',
             )
+            this.callbacks.onConnectionError()
```

Comment:
Do you have a handler for when the connection is interrupted after it's already connected?

Reply:
there is only a single callback on the EventSource library, and it doesn't distinguish between them: 
<img width="637" alt="image" src="https://github.com/DevCycleHQ/js-sdks/assets/1219069/c50b4dd5-d6fd-4d9d-a114-f45e03b88196">

Reply:
I think you might be flapping a lot if you're not doing some sort of error handling checks to make sure the connection is actually dead